pkg/algorithm/nodeselect/v1alpha1/select_node.go

```diff
@@ -161,13 +163,19 @@ func (ac *Config) selectNode(nodeBlockDeviceMap map[string]*blockDeviceList) (*n
 	// node for specific pool type
 	minRequiredDiskCount := spcv1alpha1.DefaultDiskCount[ac.poolType()]
 
+	defer func() {
+		if err == nil {
+			ac.VisitedNodes[selectedBlockDevice.NodeName] = true
```

Comment:
will this work even if NodeName is empty ""?
This can happen if nothing from nodeBlockDeviceMap is selected.

Reply:
> this work even if NodeName is empty ""?

Yes it will work but removed `defer()`